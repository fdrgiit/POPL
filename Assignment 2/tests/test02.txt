expr term factor #
PLUS MINUS MULT DIV LPAREN RPAREN NUM REALNUM #
expr -> term PLUS expr #
expr -> term MINUS expr #
expr -> term #
term -> factor MULT term #
term -> factor DIV term #
term -> factor #
factor -> LPAREN expr RPAREN #
factor -> NUM #
factor -> REALNUM # ##
These are extra input that should be ignored!
