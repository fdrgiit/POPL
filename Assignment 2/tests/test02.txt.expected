FIRST(expr) = { LPAREN, NUM, REALNUM }
FIRST(term) = { LPAREN, NUM, REALNUM }
FIRST(factor) = { LPAREN, NUM, REALNUM }
FOLLOW(expr) = { $, RPAREN }
FOLLOW(term) = { $, MINUS, PLUS, RPAREN }
FOLLOW(factor) = { $, DIV, MINUS, MULT, PLUS, RPAREN }
